lib/chef/provider.rb

```diff
@@ -204,26 +204,39 @@ def converge_if_changed(*properties, &converge_block)
         specified_properties = properties.select { |property| new_resource.property_is_set?(property) }
         modified = specified_properties.select { |p| new_resource.send(p) != current_resource.send(p) }
         if modified.empty?
-          Chef::Log.debug("Skipping update of #{new_resource.to_s}: has not changed any of the specified properties #{specified_properties.map { |p| "#{p}=#{new_resource.send(p).inspect}" }.join(", ")}.")
```

Comment:
For this one, can you change the `if sensitive` branch to just not print the values? Like this:

```ruby
if sensitive
  properties_str = specified_properties.map { |p| "#{p}=#{new_resource.send(p).inspect" }.join(", ")
else
  properties_str = specified_properties.join(", ")
end
```

This way, you still get the useful information about what properties changed, just not what their values were.